launchpad/api.go

```diff
@@ -0,0 +1,98 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package launchpad
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/z"
+	"github.com/obolnetwork/charon/cluster"
+)
+
+// New returns a new Client.
+func New(url string) Client {
+	return Client{
+		baseURL: url,
+	}
+}
+
+// Client is the REST client for launchpad API requests.
+type Client struct {
+	baseURL string // Base launchpad URL
+}
+
+// PublishLock posts the lockfile to obol-api.
+func (c Client) PublishLock(ctx context.Context, lock cluster.Lock) error {
+	if lock.Version != "v1.5.0" {
```

Comment:
what happens to future lock versions, say `v1.6.0`? does this check need to be updated then?

Reply:
suggest not checking version, the server can check if it supports it and error if it doesn't